main.go

```diff
@@ -0,0 +1,192 @@
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(
+		mongoUri,
+	))
+
+	defer func() {
+		if err = client.Disconnect(ctx); err != nil {
+			panic(err)
+		}
+	}()
+
+	err = client.Ping(ctx, nil)
+
+	if err != nil {
+		fmt.Println("There was a problem connecting to your Atlas cluster. Check that the URI includes a valid username and password, and that your IP address has been added to the access list. Error: ")
+		panic(err)
+	}
+
+	fmt.Println("Connected to MongoDB!")
+
+	// Provide the name of the database and collection you want to use.
+	// If they don't already exist, the driver and Atlas will create them
+	// automatically when you first write data.
+	var dbName = "myDatabase"
+	var collectionName = "recipes"
+	collection := client.Database(dbName).Collection(collectionName)
+
+	/*      *** INSERT DOCUMENTS ***
+	 *
+	 * You can insert individual documents using collection.Insert().
+	 * In this example, we're going to create 4 documents and then
+	 * insert them all in one call with InsertMany().
+	 */
+
+	eloteRecipe := Recipe{
+		Name:              "elote",
+		Ingredients:       []string{"corn", "mayonnaise", "cotija cheese", "sour cream", "lime"},
+		PrepTimeInMinutes: 35,
+	}
+
+	locoMocoRecipe := Recipe{
+		Name:              "loco moco",
+		Ingredients:       []string{"ground beef", "butter", "onion", "egg", "bread bun", "mushrooms"},
+		PrepTimeInMinutes: 54,
+	}
+
+	patatasBravasRecipe := Recipe{
+		Name:              "patas bravas",
+		Ingredients:       []string{"potato", "tomato", "olive oil", "onion", "garlic", "paprika"},
+		PrepTimeInMinutes: 80,
+	}
+
+	friedRiceRecipe := Recipe{
+		Name:              "fried rice",
+		Ingredients:       []string{"rice", "soy sauce", "egg", "onion", "pea", "carrot", "sesame oil"},
+		PrepTimeInMinutes: 40,
+	}
+
+	// Create an interface of all the created recipes
+	recipes := []interface{}{eloteRecipe, locoMocoRecipe, patatasBravasRecipe, friedRiceRecipe}
+	insertManyResult, err := collection.InsertMany(context.TODO(), recipes)
+	if err != nil {
+		fmt.Println("Something went wrong trying to insert the new documents:")
+		panic(err)
+	}
+	fmt.Println("the following documents were inserted:", insertManyResult.InsertedIDs)
+
+	/*
+	 * *** FIND DOCUMENTS ***
+	 *
+	 * Now that we have data in Atlas, we can read it. To retrieve all of
+	 * the data in a collection, we create a filter for recipes that take
+	 * less than 45 minutes to prepare and sort by name (ascending)
+	 */
+
+	var filter = bson.M{"prepTimeInMinutes": bson.M{"$lt": 45}}
+	options := options.Find()
+
+	// Sort by `name` field ascending
+	options.SetSort(bson.D{{"name", 1}})
+
+	cursor, err := collection.Find(context.TODO(), filter, options)
+	if err != nil {
+		fmt.Println("Something went wrong trying to find the documents:")
+		panic(err)
+	}
+
+	defer func() {
+		cursor.Close(context.Background())
```

Comment:
Question: is there a specific reason for using `context.Background()` instead of `context.TODO()`? Just curious for my own edification.

Reply:
The godocs use context.Background: https://godoc.org/go.mongodb.org/mongo-driver/mongo
<img width="815" alt="Screen Shot 2020-07-14 at 11 53 28 AM" src="https://user-images.githubusercontent.com/52428905/87447512-acb69780-c5c8-11ea-9b9d-d7c3ab06b4c9.png">